src/dict/dsl_details.cc

```diff
@@ -819,47 +819,45 @@ DslScanner::DslScanner( string const & fileName ):
   if ( !f )
     throw exCantOpen( fileName );
 
-  // Now try guessing the encoding by reading the first two bytes
+  // Now try guessing the encoding
 
-  unsigned char firstBytes[ 50 ];
+  constexpr size_t firstBytesSize = 50;
+  unsigned char firstBytes[ firstBytesSize ];
 
-  if ( gzread( f, firstBytes, sizeof( firstBytes ) ) != sizeof( firstBytes ) ) {
+  if ( gzread( f, firstBytes, firstBytesSize ) != firstBytesSize ) {
     // Apparently the file's too short
     gzclose( f );
     throw exMalformedDslFile( fileName );
   }
 
   bool needExactEncoding = false;
 
-  QByteArray ba = QByteArray::fromRawData( (const char *)firstBytes, 50 );
-  codec         = QTextCodec::codecForUtfText( ba, nullptr );
-  if ( !codec ) {
-    // the encoding has no bom.
-    // check the first char # (0x23).
-    auto hashTag = 0x0023;
-
-    auto uci = qFromUnaligned< uint32_t >( firstBytes );
-    if ( uci == qToBigEndian( hashTag ) ) {
-      codec = QTextCodec::codecForMib( 1018 ); // utf-32 be
-    }
-    else if ( uci == qToLittleEndian( hashTag ) ) {
-      codec = QTextCodec::codecForMib( 1019 ); // utf-32 le
-    }
-    else {
-      auto uc = qFromUnaligned< uint16_t >( firstBytes );
-      if ( uc == qToBigEndian( uint16_t( hashTag ) ) ) {
-        codec = QTextCodec::codecForMib( 1013 ); // utf16 be
-      }
-      else if ( uc == qToLittleEndian( uint16_t( hashTag ) ) ) {
-        codec = QTextCodec::codecForMib( 1014 ); // utf16 le
-      }
-      else {
-        //default encoding
-        codec = QTextCodec::codecForName( "UTF-8" );
-      }
+  // Note that .dsl format always starts with "#NAME"
+  if ( auto guessedEncoding = QStringConverter::encodingForData( { firstBytes, firstBytesSize }, '#' );
+       guessedEncoding.has_value() ) {
+    switch ( guessedEncoding.value() ) {
+      case QStringConverter::Utf8:
+        encoding = Utf8::Utf8;
+        break;
+      case QStringConverter::Utf16LE:
+        encoding = Utf8::Utf16LE;
+        break;
+      case QStringConverter::Utf16BE:
+        encoding = Utf8::Utf16BE;
+        break;
+      case QStringConverter::Utf32LE:
+        encoding = Utf8::Utf16LE;
+        break;
+      case QStringConverter::Utf32BE:
+        encoding = Utf8::Utf32BE;
```

Comment:
what about to get rid of the Utf8 enum and use QStringConverter's enum instead?

Reply:
If we ever want to unify encoding names, more entries than QStringConverter's enum are needed (e.g. Windows-1251).

So, no.

Reply:
Maybe we should have an internal one -> `enum encodingNameIANA`.

https://www.iana.org/assignments/character-sets/character-sets.xhtml